### PR TITLE
[user-authn] Add openapi spec

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Description
Add the `openapi` folder with specs that describes Deckhouse configmap and Helm templates inputs. 

## Why do we need it, and what problem does it solve?
This is a part of Global refactoring. We need these specs to validate values coming from users.

## Changelog entries
```changes
module: user-authn
type: feature
description: Add an OpenAPI spec to validate Deckhouse configuration parameters for the user-authn module.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "deploy/web/test"...

**Review by z9r5 — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "deploy/web/stage"...

**Comment by deckhouse-BOaTswain** [comment]
Run workflow "deploy/web/stage"...

File: modules/150-user-authn/openapi/config-values.yaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            A couple of words how this authentication metod differs from others.
+        masterCA:
+          type: string
+          description: |
+            A CA for accessing the API:
+            * If the parameter is not set, Kubernetes CA is used.
+            * We recommend using a self-signed certificate (and specify it as masterCA) if an HTTP proxy (that terminates HTTPS traffic) is used for exposing.
+  idTokenTTL:
+    type: string
+    pattern: '^[0-9]+[mh]{1}$'
```
Comment (yalosev) [inline modules/150-user-authn/openapi/config-values.yaml:87]: '^[0-9]+[mh]{1}$'
```suggestion
    pattern: '^[0-9]+[smh]{1}$'
```

**Review by yalosev — CHANGES_REQUESTED** [review]

**Review by z9r5 — APPROVED** [review]

**Pull request merged**